### DISCUSSION TITLE
Username case sensitive

### DISCUSSION THREAD

**Discussion by zelytra** [keycloak/keycloak#28492] [Q&A] [answered]
Hello,

I'm exploring the possibility of making usernames case-sensitive within my Keycloak client. I noticed that this topic has sparked extensive debate within the authentication community, particularly in discussion #10198. From what I've gathered, implementing case sensitivity may involve storing multiple forms of a username. Could anyone provide guidance on how to approach this? Is it possible through the use of a mapper, and if so, how can it be implemented effectively?

**Comment by codespearhead** [comment, marked as answer]
It seems that this question was [correctly answered in that issue](https://github.com/keycloak/keycloak/discussions/10198#discussioncomment-9045903).

In that case, can you close this discussion?
